### PR TITLE
2725 - ClickAwayListener reworked

### PR DESCRIPTION
**Pull request by Fwerfad**

### CODE REVIEW

File: jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ClickAwayListenerPage.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class ClickAwayListenerPage extends WebPage {
+
+    @UI("//h2[text()='Example']/following-sibling::div[1]/div/button")
+    public Button exampleButton;
+
+    @UI("//h2[text()='Portal']/following-sibling::div[1]/div/button")
+    public Button portalButton;
+
+    @UI("//div[text()='Click me, I will stay visible until you click outside.']")
+    public TextArea text;
```
Comment (evasileva-epam) [inline jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ClickAwayListenerPage.java:17]: Please do locators static and use it in code without "ClickAwayListenerPage." at the beginning

**Review by evasileva-epam — CHANGES_REQUESTED** [review]

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ClickAwayListenerTests.java [hunk 2]
```diff
@@ -2,66 +2,59 @@
...
-        textFieldTest();
+        textFieldTest(2);
     }
 
-    private void textFieldTest() {
-        clickAwayListenerFrame.openMenuButton.click();
-        clickAwayListenerFrame.textArea.is().displayed();
+    private void textFieldTest(int buttonId) {
+        clickAwayListenerPage.clickButton(buttonId);
+        clickAwayListenerPage.text.is().displayed();
```
[the remark this answers is not in the document]
Comment (Fwerfad) [reply jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ClickAwayListenerTests.java:35]: It's already checked in textfield locator

File: jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ClickAwayListenerPage.java (hunk 1 above)
Comment (Fwerfad) [reply jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ClickAwayListenerPage.java]: done

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ClickAwayListenerTests.java [hunk 3]
```diff
@@ -2,66 +2,59 @@
...
         //initializing all border points of a text field
```
[the remark this answers is not in the document]
Comment (Fwerfad) [reply jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ClickAwayListenerTests.java]: done

[the remark this answers is not in the document]
Comment (Fwerfad) [reply jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ClickAwayListenerTests.java]: done

**Pull request merged**